### PR TITLE
add global contact search

### PR DESCRIPTION
**Pull request by futurepaul**
uses primal's search api to search everybody on nostr

when testing on signet you can search mutinynet and you get the refund address

clicking a search result will go to the send screen and add that user to your contacts

### CODE REVIEW

**Comment by futurepaul** [comment]
https://global-search.mutiny-web.pages.dev/

File: src/utils/fetchZaps.ts [hunk 1]
```diff
@@ -316,3 +316,60 @@ export const fetchNostrProfile: ResourceFetcher<
...
+            try {
+                const content = JSON.parse(object.content);
+                const contact: Partial<PseudoContact> = {
+                    hexpub: object.pubkey
+                };
+                if (content?.name) {
+                    contact.name = content.name;
+                } else if (content?.display_name) {
+                    contact.name = content.display_name;
+                }
```
Comment (benthecarman) [inline src/utils/fetchZaps.ts:360]: in mutiny-node we do display_name then try name, should have the logic the same here

**Review by benthecarman — CHANGES_REQUESTED** [review]

**Comment by benthecarman** [comment]
this only searches by username, would be nice if it also worked with npubs

File: src/utils/fetchZaps.ts (hunk 1 above)
Comment (futurepaul) [reply src/utils/fetchZaps.ts:360]: yeah let's unify this somewhere but not now

Comment (benthecarman) [reply src/utils/fetchZaps.ts:360]: In damus my `display_name` is Carman 🏴‍☠ but my `name` is `@benthecarman` so I definitely think we should use `display_name` here

Comment (futurepaul) [reply src/utils/fetchZaps.ts]: find you convinced me

**Pull request merged**